src/Server/Provision/ServerProvisioner.php

```diff
@@ -163,26 +163,35 @@ public function check()
      */
     private function checkEntities()
     {
-        if ($entities = $this->getEntities()) {
-            return $entities;
+        $isCheck = true;
+        if ($entities = $this->getEntities($isCheck)) {
+            return $entities[0];
         }
 
         throw new \Exception('No matching Entity found in inventory');
     }
 
     /**
-     * @return Entity|void
+     * @param boolean $isCheck
+     *
+     * @return array|void
      */
-    public function getEntities()
+    public function getEntities($isCheck = false)
     {
-        return Entity::query()
+        $ipGroups = $this->ipGroupChoices();
+
+        $query = Entity::query()
             ->where('group', [
-                'billing' => $this->ipGroupChoice(),
+                'billing' => $ipGroups,
             ])
             ->where('billing_id', $this->getIps())
-            ->where('server', 'none')
-            ->first()
-            ;
+            ->where('server', 'none');
+
+        if ($isCheck || count($ipGroups) == 1) {
+            return (array)$query->first();
```

Comment:
this will return `[null]` in the case of a missing entity, which will pass the `checkEntities` check but still represent a missing Entity. We should use:
```suggestion
            return array_filter([$query->first()]);
```